src/Export/ProductDebugService.php

```diff
@@ -0,0 +1,204 @@
+
+        if (!$isExported) {
+            $this->checkExportCriteria();
+        }
+
+        return new JsonResponse([
+            'export' => [
+                'productId' => $this->product->getId(),
+                'exportedMainProductId' => $exportedMainProductId,
+                'isExported' => $isExported,
+                'reasons' => array_merge($this->parseExportErrors(), $this->reasons)
+            ],
+            'debugLinks' => [
+                'exportUrl' => $this->buildExportUrl($exportedMainProductId),
+                'debugUrl' => $this->buildDebugUrl($exportedMainProductId)
+            ],
+            'data' => [
+                'isExportedMainVariant' => $exportedMainProductId === $this->product->getId(),
+                'product' => $this->product,
+                'siblings' => $this->product->getParentId() ? $this->getSiblings($this->product) : [],
+                'associations' => $this->buildCriteria()->getAssociations(),
+            ]
+        ]);
+    }
+
+    public function fetchProduct(?string $productId = null, ?bool $withVariantInformation = false): ?ProductEntity
+    {
+        $criteria = $this->buildCriteria($productId);
+
+        /** @var EntitySearchResult $entityResult */
+        $entityResult = $this->getContainer()->get('product.repository')->search(
+            $criteria,
+            $this->getSalesChannelContext()->getContext()
+        );
+
+        return $withVariantInformation
+            ? $this->buildProductsWithVariantInformation($entityResult)->first()
+            : $entityResult->first();
+    }
+
+    private function searchProduct(Criteria $criteria): ?ProductEntity
+    {
+        return $this->getContainer()->get('product.repository')->search(
+            $criteria,
+            $this->getSalesChannelContext()->getContext()
+        )->first();
+    }
+
+    private function buildCriteria(?string $productId = null, ?bool $withAssociations = true): Criteria
+    {
+        $criteria = new Criteria([$productId ?? $this->productId]);
+
+        if ($withAssociations) {
+            Utils::addProductAssociations($criteria);
+        }
+
+        return $criteria;
+    }
+
+    private function exportedMainVariantId(): string
+    {
+        $product = $this->fetchProduct($this->productId, true);
+
+        return $product->getId();
+    }
+
+    private function isExported(): bool
+    {
+        $result = $this->searchVisibleProducts(1, 0, $this->productId);
+
+        if ($result->count() === 1) {
+            if ($result->first()->getId() === $this->productId) {
+                return true;
+            } else {
+                $this->reasons[] = 'Different variant is used in the export.';
+            }
+        } else {
+            $this->reasons[] = 'Product could not be found or is not available for search.';
+        }
+
+        return false;
+    }
+
+    private function checkExportCriteria(): void
+    {
+        $criteriaFunctions = [
```

Comment:
In this context, maybe `methods` make more sense than `functions`. I personally think of a function more as a callback or something that is defined inline, methods are usually strictly bound to class instances.